daemon/snap.go

```diff
@@ -99,6 +99,9 @@ func allLocalSnapInfos(st *state.State) ([]aboutSnap, error) {
 	for name, snapState := range snapStates {
 		info, err := snap.ReadInfo(name, snapState.CurrentSideInfo())
 		if err != nil {
+			if _, ok := err.(*snap.SnapVanishedError); ok {
```

Comment:
This branch looks fine as a minor step forward, but it's not clear if we should sprinkle the brittleness of this logic throughout the code.

What is the real solution, and when can we have it?  We need a place where we can find out about the broken state, and clean it up.